opensrp-anc/src/main/java/org/smartregister/anc/library/activity/MainContactActivity.java

```diff
@@ -173,11 +165,36 @@ private void initializeMainContactContainers() {
 
     }
 
+    private void setRequiredFields(Contact contact) {
+        if (requiredFieldsMap.containsKey(contact.getName())) {
+            contact.setRequiredFields(requiredFieldsMap.get(contact.getName()));
+        }
+    }
+
+    @Override
+    public void startForms(View view) {
+        Contact contact = (Contact) view.getTag();
+        try {
+            if (contact != null) {
+                loadContactGlobalsConfig();
+                process(contactForms);
+
+                setRequiredFields(contact);
+                view.setTag(contact);
+                super.startForms(view);
+            }
+        } catch (IOException e) {
+            Timber.e(e, " --> startForms");
+        } catch (Exception e) {
+            Timber.e(e, " --> startForms");
```

Comment:
You are catching both `Exception` and `IOException` one is the child of another, kindly fix it